Add option to allow lowercase first words in headings

Bugg4 opened the pull request:
This adds the `allowFirstWordLowerCase` boolean option, which behaves as follows:
 - `true`: Allows the first word of a heading to be lowercase, but **only** if that word is explicitly listed in your custom `lowerCaseWords` array.
  - `false`: The first word is always capitalized unless it is already all uppercase (same as previous behavior)

Resolves #19

Bugg4 committed:
Add option to allow lowercase first words in headings

ilyatitovich commented:
Hi! Thanks for the PR. Can you add tests to make sure everything works as expected?

Bugg4 committed:
Add tests for `allowFirstWordLowerCase` option

Bugg4 committed:
Tag tests

This adds a tag in front of test descriptions so that first glance it's clear what option / feature they're testing.

Bugg4 commented:
Let me know if any other tests should be added

Bugg4 committed:
Update test

Bugg4 committed:
Empty cache before running each test.

This was necessary because, when running two tests that process the same source but **with a boolean option flipped**, cache persisteance made it so that the title wasn't processed, while the test expected result changed.

Bugg4 committed:
Remove stray logging

Bugg4 committed:
Small fix

Bugg4 commented:
Closing since not needed anymore (https://github.com/ilyatitovich/remark-lint-heading-capitalization/issues/19#issuecomment-3177773176)
